migration/idempotent/001-base.sql

```diff
@@ -3026,45 +3142,49 @@ BEGIN
         FROM _prom_catalog.get_metrics_that_need_compression()
     LOOP
         IF NOT _prom_catalog.lock_metric_for_maintenance(r.id, wait=>false) THEN
-            remaining_metrics := remaining_metrics || r;
+            remaining_metrics := remaining_metrics OPERATOR(pg_catalog.||) r;
             CONTINUE;
         END IF;
         IF log_verbose THEN
-            startT := clock_timestamp();
+            startT := pg_catalog.clock_timestamp();
             RAISE LOG 'promscale maintenance: compression: metric %: starting, without lock wait', r.metric_name;
         END IF;
-        PERFORM _prom_catalog.set_app_name( format('promscale maintenance: compression: metric %s', r.metric_name));
+        PERFORM _prom_catalog.set_app_name( pg_catalog.format('promscale maintenance: compression: metric %s', r.metric_name));
         CALL _prom_catalog.compress_metric_chunks(r.metric_name);
         IF log_verbose THEN
-            RAISE LOG 'promscale maintenance: compression: metric %: finished in %', r.metric_name, clock_timestamp()-startT;
+            RAISE LOG 'promscale maintenance: compression: metric %: finished in %', r.metric_name, pg_catalog.clock_timestamp() OPERATOR(pg_catalog.-) startT;
         END IF;
         PERFORM _prom_catalog.unlock_metric_for_maintenance(r.id);
 
         COMMIT;
+        -- reset search path after transaction end
+        SET LOCAL search_path = pg_catalog;
     END LOOP;
 
     FOR r IN
         SELECT *
         FROM unnest(remaining_metrics)
```

Comment:
schema qualify unnest?

Reply:
As discussed on slack, it's not necessary to set `search_path` and schema qualify. For readability I've explicitly done either one or the other.